### PR TITLE
[ADD][9.0] project_task_send_by_mail module

### PR DESCRIPTION
**Pull request by cubells**
This module allows user to send project task by email. 

cc @Tecnativa

### CODE REVIEW

File: project_task_send_by_mail/__openerp__.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    "name": "Project Task Send By Mail",
+    "version": "9.0.1.0.0",
+    "author": "Tecnativa, "
+               "Odoo Community Association (OCA)",
+    "website": "https://www.tecnativa.com",
+    "category": "Project Management",
+    "license": "AGPL-3",
+    "depends": [
+        "project",
+        "mail",
```
Comment (pedrobaeza) [inline project_task_send_by_mail/__openerp__.py]: No need to depend on mail. Depends on project_task_report.

File: project_task_send_by_mail/data/task_template.xml [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+<odoo noupdate="1">
+
+    <record id="email_task_template" model="mail.template">
+        <field name="name">Task Order - Send by Email</field>
+        <field name="email_from">${(object.user_id.email and '%s &lt;%s&gt;' % (object.user_id.name, object.user_id.email) or '')|safe}</field>
+        <field name="subject">${object.company_id.name} Task (Ref ${object.name or 'n/a'})</field>
+        <field name="partner_to">${object.partner_id.id}</field>
+        <field name="model_id" ref="project.model_project_task"/>
+        <field name="auto_delete" eval="True"/>
+        <field name="report_template" ref="project_task_send_by_mail.report_taskorder"/>
```
Comment (pedrobaeza) [inline project_task_send_by_mail/data/task_template.xml]: Update this to project_task_report XML-ID

**Review by pedrobaeza — CHANGES_REQUESTED** [review]

File: project_task_send_by_mail/README.rst [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+you to send the task order by email.
+
+Usage
+=====
+
+To use this module, you need to:
+
+* Go to a Task Order.
+* Click on the button Send by Email.
+* Select the template: Task Order - Send by Email.
```
Comment (pedrobaeza) [inline project_task_send_by_mail/README.rst]: Remove this.

File: project_task_send_by_mail/models/project.py [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+        )
+        compose_form = self.env.ref('mail.email_compose_message_wizard_form',
+                                    False)
+        ctx = dict(
+            default_model='project.task',
+            default_res_id=self.id,
+            default_use_template=bool(template),
+            default_template_id=template and template.id or False,
+            default_composition_mode='comment',
+            mark_invoice_as_sent=True,
```
Comment (pedrobaeza) [inline project_task_send_by_mail/models/project.py]: Remove this key as useless

**Review by pedrobaeza — CHANGES_REQUESTED** [review]

**Review by pedrobaeza — APPROVED** [review]
@rafaelbn @chienandalu please review

**Review by chienandalu — CHANGES_REQUESTED** [review]
The mail compose windows attaches Task.pdf wich actually is an empty plain text file.

**Review by rafaelbn — CHANGES_REQUESTED** [review]
- The PDF generated is broken and PDF reader cannot open it

**Comment by rafaelbn** [comment]
@cubells all looks fine, just the PDF file is not valid, try to open this PDF:

http://3299908-303-b81f1c.runbot2.odoo-community.org/web?reload#id=1&view_type=form&model=project.task&menu_id=192&action=256

**Comment by chienandalu** [comment]
@rafaelbn @cubells Seems that there's an issue with this wkhtmltopdf:

image

**Comment by pedrobaeza** [comment]
The problem with Runbot printing reports persists.

@cubells try to remove all as the branch that is working.

**Review by chienandalu — APPROVED** [review]
Approve as wkhtmltopdf is not related

**Comment by pedrobaeza** [comment]
@rafaelbn try now in http://3299995-303-1b1e3a.runbot2.odoo-community.org/web that is working correctly

**Review by rafaelbn — APPROVED** [review]
Tested 👍

**Pull request merged**